src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -900,68 +922,95 @@ std::string UsgsAstroFrameSensorModel::constructStateFromIsd(const std::string&
       state["m_currentParameterValue"][2] = 0;
     }
     else {
-      state["m_currentParameterValue"] = position;
+      state["m_currentParameterValue"] = positions;
     }
 
     // get sensor_velocity
-    std::vector<double> velocity = getSensorVelocities(isd, parsingWarnings);
-    if (!velocity.empty() && velocity.size() != 3) {
+    if (!velocities.empty() && velocities.size() != 3) {
       parsingWarnings->push_back(
         csm::Warning(
           csm::Warning::DATA_NOT_AVAILABLE,
           "Sensor velocity does not have 3 values.",
           "UsgsAstroFrameSensorModel::constructStateFromIsd()"));
     }
     else {
-      state["m_spacecraftVelocity"] = velocity;
+      state["m_spacecraftVelocity"] = velocities;
     }
 
 
     // get sun_position
     // sun position is not strictly necessary, but is required for getIlluminationDirection.
-    state["m_sunPosition"] = getSunPositions(isd);
+    ale::States sunState = ale::getSunPosition(parsedIsd);
+    std::vector<ale::State> sunStates = sunState.getStates();
+    ephemTime = sunState.getTimes();
+    ale::State rotatedSunState;
+    std::vector<double> sunPositions = {};
+
+    for (int i = 0; i < ephemTime.size(); i++) {
+      rotatedSunState = j2000_to_target.rotateStateAt(ephemTime[i], sunStates[i]);
+      sunPositions.push_back(rotatedSunState.position.x * 1000);
+      sunPositions.push_back(rotatedSunState.position.y * 1000);
+      sunPositions.push_back(rotatedSunState.position.z * 1000);
+    }
+
+    state["m_sunPosition"] = sunPositions;
 
     // get sensor_orientation quaternion
-    std::vector<double> quaternion = getSensorOrientations(isd, parsingWarnings);
-    if (quaternion.size() != 4) {
+    ale::Orientations j2000_to_sensor = ale::getInstrumentPointing(parsedIsd);
+    ale::Orientations sensor_to_j2000 = j2000_to_sensor.inverse();
+    ale::Orientations sensor_to_target = j2000_to_target * sensor_to_j2000;
+    ephemTime = sensor_to_target.getTimes();
+    std::vector<double> quaternion;
+    std::vector<double> quaternions;
+
+    for (int i = 0; i < ephemTime.size(); i++) {
+      ale::Rotation rotation = sensor_to_target.interpolate(ephemTime[i], ale::SLERP);
+      quaternion = rotation.toQuaternion();
+      quaternions.push_back(quaternion[1]);
```

Comment:
Naming is a little confusing. This is still just one quaternion, but moving from w-first to w-last format, right?